src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,240 @@
+import {
+  Directive,
+  forwardRef,
+  ElementRef,
+  EventEmitter,
+  Input,
+  HostListener,
+  AfterViewInit,
+  Injector,
+  ChangeDetectorRef,
+  OnInit,
+  Output,
+  OnDestroy,
+  Optional
+} from '@angular/core';
+
+import {
+  NG_VALUE_ACCESSOR,
+  NG_VALIDATORS,
+  ControlValueAccessor,
+  Validator,
+  AbstractControl,
+  NgControl,
+  FormControl
+} from '@angular/forms';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import {
+  SkyCountryData
+} from './types';
+
+import {
+  SkyPhoneFieldAdapterService
+} from './phone-field-adapter.service';
+
+import {
+  SkyPhoneFieldComponent
+} from './phone-field.component';
+
+// tslint:disable:no-forward-ref no-use-before-declare
+const SKY_PHONE_FIELD_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+
+const SKY_PHONE_FIELD_VALIDATOR = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+// tslint:enable
+
+@Directive({
+  selector: '[skyPhoneFieldInput]',
+  providers: [
+    SKY_PHONE_FIELD_VALUE_ACCESSOR,
+    SKY_PHONE_FIELD_VALIDATOR,
+    SkyPhoneFieldAdapterService
+  ]
+})
+export class SkyPhoneFieldInputDirective implements OnInit, OnDestroy, AfterViewInit,
+  ControlValueAccessor, Validator {
+
+  @Input()
+  public set defaultCountry(value: string) {
+    this.skyPhoneFieldComponent.setDefaultCountry(value);
+    this._defaultCountryCode = value;
+  }
+
+  public get defaultCountry() {
+    return this._defaultCountryCode;
+  }
+
+  @Input()
+  public set disabled(value: boolean) {
+    this.skyPhoneFieldComponent.disabled = value;
+    this.adapterService.setElementDisabledState(this.elRef.nativeElement, value);
+    this._disabled = value;
+  }
+
+  public get disabled(): boolean {
+    return this._disabled || false;
+  }
+
+  @Input()
+  public formatModel: boolean = true;
+
+  @Input()
+  public skyPhoneFieldNoValidate: boolean = false;
+
+  @Output()
+  public selectedCountryChange = new EventEmitter<SkyCountryData>();
+
+  private set modelValue(value: string) {
+    this._modelValue = value;
+    this.adapterService.setElementValue(this.elRef.nativeElement, value);
+    if (value) {
```

Comment:
The spacing feels a bit claustrophobic in this setter. Ways we can clean it up to make the logic more readable?

Reply:
Made some changes here. Any better now?

Reply:
Much!